pkg/machinery/config/types/v1alpha1/generate/generate.go

```diff
@@ -439,14 +445,26 @@ func NewInput(clustername, endpoint, kubernetesVersion string, secrets *SecretsB
 
 	additionalSubjectAltNames = append(additionalSubjectAltNames, options.AdditionalSubjectAltNames...)
 
+	clusterID, err := genV4UUID()
```

Comment:
this is what I talked about yesterday: this values are cluster-wide secrets and they should be put to the `Secrets` struct so that when we join the nodes to the existing cluster, they get same value, not a random one.

Reply:
I've never even noticed the `Secrets` struct.

Reply:
Now happily ensconced within Secrets.